central/role/datastore/telemetry.go

```diff
@@ -22,9 +25,84 @@ var Gather phonehome.GatherFunc = func(ctx context.Context) (map[string]any, err
 	rs := Singleton()
 
 	gatherErrs := errorhelpers.NewErrorList("cannot gather from role store")
-	gatherErrs.AddError(phonehome.AddTotal(ctx, totals, "PermissionSets", rs.CountPermissionSets))
-	gatherErrs.AddError(phonehome.AddTotal(ctx, totals, "Roles", rs.CountRoles))
-	gatherErrs.AddError(phonehome.AddTotal(ctx, totals, "Access Scopes", rs.CountAccessScopes))
+	gatherErrs.AddError(totalPermissionSets(ctx, totals, rs))
+	gatherErrs.AddError(totalRoles(ctx, totals, rs))
+	gatherErrs.AddError(totalAccessScopes(ctx, totals, rs))
 
 	return totals, gatherErrs.ToError()
 }
+
+func totalPermissionSets(ctx context.Context, props map[string]any, rs DataStore) error {
```

Comment:
`phonehome.AddTotal()` worked for all 3 types. Could we have a single function here as well?

Reply:
Not sure I understand that comment quite right, do you want a single function for counting the amount of resources by origin here, i.e.
```go
phonehome.AddTotal(ctx, props, "origin", rs.CountDeclarativeRoles)
```
or do you have something different in mind with the "single function"?